frmts/vrt/vrtdataset.cpp

```diff
@@ -1044,6 +1044,121 @@ GDALDataset *VRTDataset::OpenVRTProtocol(const char *pszSpec)
         return nullptr;
     }
 
+    // scan for subdataset/subdataset_n in tokens, close the source dataset and reopen with subdataset if found/valid
+    // make subdataset and subdataset_n mutually exclusive
+    bool bFound_subdataset = false;
+    for (int i = 0; i < aosTokens.size(); i++)
+    {
+        char *pszKey = nullptr;
+        const char *pszValue = CPLParseNameValue(aosTokens[i], &pszKey);
+
+        if (pszKey && pszValue)
+        {
+            char **papszSubdatasets = GDALGetMetadata(poSrcDS, "SUBDATASETS");
+
+            int nSubdatasets = CSLCount(papszSubdatasets);
+
+            if (EQUAL(pszKey, "subdataset"))
```

Comment:
perhaps ``subdataset_name`` for that one and ``subdataset_idx`` instead of ``subdataset_n``  for clarity ?

Reply:
I find this hard, in the API it's "component" rather than "name" (but NAME in info). I prefer "subdataset" and "subdataset_i", but I'm ok with this will just ponder a while. It's not clear what the precedents are so I'll have a bit of a look.

Reply:
> find this hard, in the API it's "component" rather than "name" (but NAME in info).

ah, yes I missed that. I'm not too opinionated on naming...

Reply:
maybe it should be "sd_name" and "sd" to be consistent with gdalinfo? 

I'm still thinking ...

Reply:
and gdalbuildvrt also has "-sd", I don't see any other cases in apps/

Reply:
ah, missed your reply - I'm going with sd_name/sd and see if the other changes pass :)